specs/eventing/test-plan/broker-lifecycle-conformance.md

```diff
@@ -0,0 +1,188 @@
+- A [broker resource](specs/eventing/test-plan/control-plane/broker-lifecycle/broker.yaml)
+- A [trigger resource that references the broker](specs/eventing/test-plan/control-plane/broker-lifecycle/trigger.yaml)
+- A [trigger resource that doesn't reference the broker](specs/eventing/test-plan/control-plane/broker-lifecycle/trigger-no-broker.yaml)
+
+
+## [Pre] Creating a Broker 
+
+```
+kubectl apply -f control-plane/broker-lifecycle/broker.yaml
+```
+
+
+## [Test] Immutability
+
+Check for default annotations, this should return the name of the selected implementation: 
+
+```
+kubectl get broker conformance-broker -o jsonpath='{.metadata.annotations.eventing\.knative\.dev/broker\.class}'
+```
+
+Try to patch the annotation: `eventing.knative.dev/broker.class` to see if the resource mutates: 
+
+```
+kubectl patch broker conformance-broker --type merge -p '{"metadata":{"annotations":{"eventing.knative.dev/broker.class":"mutable"}}}'
+```
+
+You should get the following error: 
+```
+Error from server (BadRequest): admission webhook "validation.webhook.eventing.knative.dev" denied the request: validation failed: Immutable fields changed (-old +new): annotations
+{string}:
+	-: "MTChannelBasedBroker"
+	+: "mutable"
+```
+
+### [Output]
+
+```
+{
+  "test": "control-plane/broker-lifecycle/immutability-1"
+  "output": {
+    	"brokerImplementation": "<BROKER IMPLEMENTATION>",
+	"expectedError": "<EXPECTED ERROR>"
+  }
+}
+```
+
+Try to mutate the `.spec.config` to see if the resource mutates: 
+
+```
+kubectl patch broker conformance-broker --type merge -p '{"spec":{"config":{"apiVersion":"v1"}}}'
+```
+
+
+### [Output]
+
+```
+{
+  "test": "control-plane/broker-lifecycle/immutability-2"
+  "output": {
+  	"brokerImplementation": "<BROKER IMPLEMENTATION>",
+	"expectedError": "<EXPECTED ERROR>"
+  }
+}
+```
+
+
+## [Test] Broker Readyness 
+
+Check for condition type `Ready` with status `True`: 
+
+```
+ kubectl get broker conformance-broker -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+### [Output]
+
+```
+{
+  "test": "control-plane/broker-lifecycle/broker-readyness"
+  "output": {
+  	"brokerImplementation": "<BROKER IMPLEMENTATION>",
+	"expectedType": "Ready",
+	"expectedStatus": "True"
+  }
+}
+```
+
+## [Test] Broker is Addresable
+
+Running the following command should return a URL
+
+```
+kubectl get broker conformance-broker -ojsonpath="{.status.address.url}"
+```
+
+### [Output]
+
+```
+{
+  "test": "control-plane/broker-lifecycle/broker-addressable"
+  "output": {
+  	"brokerImplementation": "",
+	"obtainedURL": "<BROKER URL>",
+  }
+}
+```
+
+## [Pre] Create Trigger for Broker
+
+Create a trigger that points to the broker:
+
+```
+kubectl apply -f control-plane/broker-lifecycle/trigger.yaml
+```
+
+## [Test] Broker Reference in Trigger
+
+Check that the `Trigger` is making a reference to the `Broker`, this should return the name of the broker.
```

Comment:
And this convention is good, with the "`" and starting with upper-case, but we should have it consistent in all the file